projects/js-packages/publicize-components/src/components/message-box-control/index.js

```diff
@@ -10,16 +12,37 @@ import { __, _n, sprintf } from '@wordpress/i18n';
  * @param {Function} props.onChange  - Callback to invoke as the message changes.
  * @param {boolean}  [props.disabled]  - Whether the control is disabled.
  * @param {number}   props.maxLength - The maximum character length of the message.
+ * @param {object}   props.analyticsData - Data for tracking analytics.
  * @returns {object} The message box component.
  */
-export default function MessageBoxControl( { message = '', onChange, disabled, maxLength } ) {
+export default function MessageBoxControl( {
+	message = '',
+	onChange,
+	disabled,
+	maxLength,
+	analyticsData = null,
+} ) {
+	const { recordEvent } = useAnalytics();
+	const [ isFirstChange, setIsFirstChange ] = useState( true );
```

Comment:
`useRef` will work better here instead of `useState` because the flag will be switched only once.

Reply:
Thanks, I'll switch to that